contracts/lib/OnchainNonMerklizedIdentityBase.sol

```diff
@@ -0,0 +1,245 @@
+        IdentityBase.initialize(_stateContractAddr);
+    }
+
+    /**
+     * @dev listUserCredentials. Get list of user credentials identifiers
+     * @param _userId user id
+     */
+    function listUserCredentials(uint256 _userId) external virtual returns (Id[] memory);
+
+    /**
+     * @dev getCredential. Get credential by user id and credential id
+     */
+    function getCredential(
+        uint256 _userId,
+        uint256 _credentialId
+    ) external virtual returns (CredentialData memory);
+
+    /**
+     * @dev processOnchainCredentialData. Build credential data from core claim and additional information
+     */
+    function processOnchainCredentialData(
+        CredentialInformation memory credentialInformation,
+        CredentialMetadata memory credentialMetadata,
+        SubjectField[] memory credentialSubject,
+        Claim memory claim
+    ) internal view returns (CredentialData memory) {
+        IssuerState memory issuerState = IssuerState({
+            rootOfRoots: identity.latestPublishedTreeRoots.rootsRoot,
+            claimsTreeRoot: identity.latestPublishedTreeRoots.claimsRoot,
+            revocationTreeRoot: identity.latestPublishedTreeRoots.revocationsRoot,
+            value: identity.latestPublishedState
+        });
+
+        IssuerData memory issuerData = IssuerData({id: identity.id, state: issuerState});
+
+        SmtLib.Proof memory mtp = identity.getClaimProof(claim.hashIndex);
+        require(mtp.existence, "Claim does not exist in issuer state");
+
+        IssuanceProof memory issuanceProof = IssuanceProof({
+            _type: "Iden3SparseMerkleTreeProof",
+            coreClaim: claim.coreClaim,
+            issuerData: issuerData,
+            mtp: mtp
+        });
+        IssuanceProof[] memory proofs = new IssuanceProof[](1);
+        proofs[0] = issuanceProof;
+
+        uint256 chainId;
+        assembly {
+            chainId := chainid()
+        }
+        return
+            CredentialData({
+                id: string(
+                    abi.encodePacked(
+                        "urn:onchain:",
+                        Strings.toString(chainId),
+                        ":",
+                        Strings.toHexString(uint160(address(this))),
+                        ":",
+                        Strings.toString(credentialMetadata.id)
+                    )
+                ),
+                context: fillContext(credentialInformation.jsonLDContextUrls),
+                _type: fillType(credentialInformation._type),
+                expirationDate: credentialMetadata.expirationDate,
+                issuanceDate: credentialMetadata.issuanceDate,
+                issuer: identity.id,
+                credentialSubject: fillCredentialSubject(
+                    credentialSubject,
+                    credentialInformation._type
+                ),
+                credentialStatus: Status({
+                    id: "/credentialStatus",
```

Comment:
Maybe leave it empty if anyway it's generated in schema-processor

Reply:
In my opinion, an empty ID or lack of ID looks dirtier than a simple id as `/credentialStatus`.
But If you want to remove the need to discuss it with @vmidyllic and @demonsh

Reply:
If id exists use id, if empty - generate.